lib/chef/knife/ssh.rb

```diff
@@ -186,8 +186,8 @@ def get_ssh_attribute(node)
           Chef::Log.debug("Using node attribute #{Chef::Config[:knife][:ssh_attribute]}")
           attribute = Chef::Config[:knife][:ssh_attribute]
         elsif node[:cloud] &&
-              node[:cloud][:public_hostname] &&
-              !node[:cloud][:public_hostname].empty?
+            node[:cloud][:public_hostname] &&
+            !node[:cloud][:public_hostname].empty?
           Chef::Log.debug("Using node attribute 'cloud[:public_hostname]' automatically as the ssh target")
```

Comment:
yeah the `&.` operator in ruby-2.3.0 might help with the long trainwreck guards as well.